packages/base/any/kernels/5.15-lts/patches/0001-feat-Add-dev-mvsw_fw_comm_debug-device-for-FW-debugg.patch

```diff
@@ -0,0 +1,680 @@
+commit 586d6dd722b331bc0500cbe0e239aef4110f8ca3
+Author: Vlad GEORGESCU <vlaggeor@amazon.com>
+Date:   Tue Aug 27 18:25:04 2024 +0000
+
+    feat: Add /dev/mvsw_fw_comm_debug device for FW debugging
+    
+    This commit allows users to send commands to the secondary CPU for debugging
+    purposes. By executing specific commands, users can inspect the current status
+    of the firmware and gather information useful for debugging firmware-related
+    issues.
+    
+    Signed-off-by: Vlad GEORGESCU <vlaggeor@amazon.com>
```

Comment:
Has this patch been sent (and reviewed) upstream?

Reply:
While the patch has undergone thorough internal reviews, including technical and security assessments, it has not been submitted or reviewed upstream yet. My understanding is that upstream review will be part of this pull request. Let me know if we need to initiate any review or process.

I want to mention that the patch was initially developed and reviewed for the 5.10 kernel version (used by us on TN48 builds). This commit represents a port of the original patch to the current github kernel version (5.15).

Reply:
Sorry for the misunderstanding. I meant upstream Linux kernel. I think the “dentOS GitHub community” lacks the resources of a proper review. Sometimes Marvell developers are around though. It’d be great if you could port it to Linux master branch and sent it upstream. From [`MAINTAINERS`](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/MAINTAINERS):

```
Taras Chornyi <taras.chornyi@plvision.eu> (supporter:MARVELL PRESTERA ETHERNET SWITCH DRIVER)
"David S. Miller" <davem@davemloft.net> (maintainer:NETWORKING DRIVERS)
Eric Dumazet <edumazet@google.com> (maintainer:NETWORKING DRIVERS)
Jakub Kicinski <kuba@kernel.org> (maintainer:NETWORKING DRIVERS)
Paolo Abeni <pabeni@redhat.com> (maintainer:NETWORKING DRIVERS)
netdev@vger.kernel.org (open list:NETWORKING DRIVERS)
linux-kernel@vger.kernel.org (open list)
```

@taraschornyiplv, was/is active in DENT though.

Reply:
Agree. I will create a new PR in https://github.com/dentproject/linux with changes from the patch.
This PR will remain open until the PR in  https://github.com/dentproject/linux is merged and after that I will remove the patch.
Is this ok ?

Reply:
That archive is quite outdated. Better use https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/.

Reply:
Hi Paul,

Indeed, Prestera driver from [net-next.git](https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/log/drivers/net/ethernet/marvell/prestera) looks indeed more recent. Last update is `2024-05-13` while the last integration in [dent-linux-5.15.y](https://github.com/dentproject/linux/commits/dent-linux-5.15.y/drivers/net/ethernet/marvell/prestera) is from `Dec 26, 2022` (see commit https://github.com/dentproject/linux/commit/a2056cbd223412b2dbe3d1c4c8e077450cd15bf2)

The difference between these 2 repositories is significant (+100 commits)
-  The version we currently use is 4.0.0, updated by @taraschornyiplv in this commit from `Dec 26, 2022`: https://github.com/dentproject/linux/commit/a2056cbd223412b2dbe3d1c4c8e077450cd15bf2
- In repo [net-next.git](https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/). Version 4.0 is added in this commit from 2021-10-29  [net: marvell: prestera: add firmware v4.0 support](https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/commit/drivers/net/ethernet/marvell/prestera?id=bb5dbf2cc64d5cfa696765944c784c0010c48ae8)

This raise some issues we need to consider and clarify before proceeding
- do we really want to bring latest Prestera driver into [dent-linux-5.15.y] (https://github.com/dentproject/linux/commits/dent-linux-5.15.y/drivers/net/ethernet/marvell/prestera)  ?
- This update will also need a FW update ?  We currently use Marvell firmware CPU version 4.3.5.038. Is this compatible with latest Prestera driver ? Has been tested ?

As this direction is raising some concerns that will take time to address: sync with Marvell, testing, bake time, PR review, etc, my proposal is to work on 2 directions.

1. Have a targeted commit in current Prestera driver (4.0)
 - using the patch in this PR, or
 - using a new PR in [dent-linux-5.15.y](https://github.com/dentproject/linux/commits/dent-linux-5.15.y/drivers/net/ethernet/marvell/prestera)
2. Start the process of upstreaming this change in [net-next.git](https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/log/drivers/net/ethernet/marvell/prestera)

Let me know you see this.
A feedback from @taraschornyiplv would be appreciated.

Thank you

Reply:
Dear @vlaggeor, thank you for such a detailed reply. I agree, that you merge/pull request should not be delayed by the upstreaming effort, and that your proposal sounds great.

Reply:
>     1. Have a targeted commit in current Prestera driver (4.0)
> 
> 
>     * using the patch in this PR, or
> 
>     * using a new PR in [dent-linux-5.15.y](https://github.com/dentproject/linux/commits/dent-linux-5.15.y/drivers/net/ethernet/marvell/prestera)

I think we should have only one of these, either upstream 5.15 + a set of patches, or a branch with changes applied, but not both.

Since we currently have the latter, I think this should be a PR in https://github.com/dentproject/linux/, not here.

Reply:
> I think we should have only one of these, either upstream 5.15 + a set of patches, or a branch with changes applied, but not both.

Please give a reason. We want to work upstream for easier maintenance in the long run, so I do not understand your comment.

Reply:
My reason is that all changes Dent has for the linux kernel should be either as patches, or as commit in a forked kernel, to have them all in the same place.

Storing changes in multiple places is confusing, and makes it hard to keep track of the changes. It also creates ambiguity about where a new change should be stored, and can cause conflicts easily if changes for the same kernel parts are submitted at different places.

This is independent of any upstream work or upstream suitability. Since this is a non bug fix change, even if it were submitted now and accepted right away, it would only be available in kernel 6.12 (or later).

The change would also not be added to 5.15, which Dent uses. So if we would want the feature, we would need to add the change ourselves anyway.

And even it it were being added, the prestera driver in Dent's linux fork is significantly different than the one in mainline linux.

Reply:
Sorry, I should have read the quote you referred to better. I think, @vlaggeor meant *either … or*, and already looked into adding it to [*dent-linux-5.15.y*](https://github.com/dentproject/linux/commits/dent-linux-5.15.y/drivers/net/ethernet/marvell/prestera).

> The change would also not be added to 5.15

Lot’s of stuff is backported nowadays, but it’d probably be better, if dentOS used more recent Linux kernels.

Reply:
I have created [PR#13](https://github.com/dentproject/linux/pull/13) in [dent-linux-5.15.y](https://github.com/dentproject/linux/commits/dent-linux-5.15.y/drivers/net/ethernet/marvell/prestera) branch.
I will close this PR in 1-2 days to allow time for feedback and review.
Thank you very much for feedback and responses.